From f6d5db0bff0750e87b89a33707c29ffbb67abf90 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Jirman?= <megi@xff.cz>
Date: Sun, 6 Feb 2022 16:08:24 +0100
Subject: [PATCH 306/515] arm64: dts: rk3399-pinephone-pro: Improve
 vcc5v0_typec description

As noted by dsimic, the original description was incorrect. Update
it to the new consensus from #pinedev chat.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 .../dts/rockchip/rk3399-pinephone-pro.dts     | 22 ++++++++++++-------
 1 file changed, 14 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
index ec4285beaafd..35d770e8305d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-pinephone-pro.dts
@@ -274,16 +274,22 @@ regulator-state-mem {
 	};
 
 	/*
-	 * This is not a regulator, but GPIO0_A6 is used to disable
-	 * VCC_SYS -> VBAT+ switch that helps boost that power path's
-	 * current carrying capacity.
+	 * This is not a regulator. GPIO0_A6 is used to force enable VCC_SYS <->
+	 * VBAT+ Q7 switch that helps boost the RK818's power path's current carrying
+	 * capacity when operating the phone from the battery in case VBUS_TYPEC
+	 * has 5V provided by the phone itself and not by the external power
+	 * supply. (this happens in USB host mode)
 	 *
-	 * GPIO0_A6 high: path is disabled no matter what
-	 * GPIO0_A6 low: path is enabled if there's 5V voltage on
-	 * VBUS_TYPEC
+	 * GPIO0_A6 low: Q7 is enabled if there's 0V on VBUS_TYPEC (phone is not
+	 *               powered from external power supply)
+	 * GPIO0_A6 high: Q7 is enabled no matter what
 	 *
-	 * GPIO0_A6 must be high when the phone is providing VBUS_TYPEC
-	 * power.
+	 * GPIO0_A6 must be high when the phone is providing VBUS_TYPEC power.
+	 * This keeps the power path boost enabled for all cases when the phone
+	 * runs from the battery.                                                                                      
+	 *
+	 * In other words, GPIO0_A6 is used to override the automatic disabling
+	 * of Q7 when VBUS_TYPEC has 5V in USB host mode.
 	 */
 	vcc5v0_typec: vcc5v0-typec-regulator {
 		compatible = "regulator-fixed";
-- 
2.35.3

